File: Leetcode 3601-3700/3681/3681.py
# Leetcode 3681: Maximum XOR of Subsequences
# https://leetcode.com/problems/maximum-xor-of-subsequences/
# Solved on 20th of December, 2025
class Solution:
    def maxXorSubsequences(self, nums: list[int]) -> int:
        """
        Calculates the maximum XOR sum of any subsequence of the given numbers.

        :param nums: A list of integers.
        :return: The maximum XOR sum achievable from any subsequence.
        """
        basis = [0] * 32
        for num in nums:
            for i in range(31, -1, -1):
                if (num >> i) & 1:
                    if basis[i] == 0:
                        basis[i] = num
                        break
                    num ^= basis[i]

        maxXor = 0
        for i in range(31, -1, -1):
            if (maxXor ^ basis[i]) > maxXor:
                maxXor ^= basis[i]

        return maxXor

File: Leetcode 3601-3700/3681/3681.java
// Leetcode 3681: Maximum XOR of Subsequences
// https://leetcode.com/problems/maximum-xor-of-subsequences/
// Solved on 20th of December, 2025
class Solution {
    /**
     * Finds the maximum XOR sum of any subsequence of the given array.
     * This is achieved by constructing a Gaussian basis (or XOR basis) for the numbers.
     * @param nums The input array of integers.
     * @return The maximum possible XOR sum of a subsequence.
     */
    public int maxXorSubsequences(int[] nums) {
        int[] basis = new int[32];
        for (int num : nums) {
            for (int i = 30; i >= 0; i--) {
                if ((num & (1 << i)) != 0) {
                    if (basis[i] == 0) {
                        basis[i] = num;
                        break;
                    }
                    num ^= basis[i];
                }
            }
        }

        int max = 0;
        for (int i = 30; i >= 0; i--) {
            if ((max ^ basis[i]) > max) {
                max ^= basis[i];
            }
        }
        return max;
    }
}